src/core/midiMsg.h

```diff
@@ -0,0 +1,50 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Giada - Your Hardcore Loopmachine
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2020 Giovanni A. Zuliani | Monocasual
+ *
+ * This file is part of Giada - Your Hardcore Loopmachine.
+ *
+ * Giada - Your Hardcore Loopmachine is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General
+ * Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Giada - Your Hardcore Loopmachine is distributed in the hope that it
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Giada - Your Hardcore Loopmachine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------- */
+
+
+#ifndef G_MIDIMSG_H
+#define G_MIDIMSG_H
+
+#include <vector>
+#include <string>
+
+namespace giada {
+namespace m {
+
+class midiMsg
```

Comment:
Class names should start with a capital letter (e.g. `MidiMsg`).

We already have a `MidiEvent` class (https://github.com/monocasual/giada/blob/master/src/core/midiEvent.h) that might overlap with this one. Do you think they could be merged together? Or maybe the new `MidiMsg` class will be expanded in the future in a different direction?

Reply:
> Class names should start with a capital letter (e.g. `MidiMsg`).

Good point. Let's leave this conversation open so I won't forget about this one.

> Do you think they could be merged together? Or maybe the new `MidiMsg` class will be expanded in the future in a different direction?

MidiEvent will coexist during the transition and eventually (Stage 4 I think?) will be completely replaced by more general MidiMsg + MidMsgFilter classes. MidiEvent is designed to handle only a small subset of the MIDI standard messages, and MidiMsg is supposed to handle all possible messages.

However I think MidiMsg might inherit function names, like getChannel(), getNote() etc. MidMsg of course will expand in the future, according to our needs.